src/sim65/main.c

```diff
@@ -140,6 +141,10 @@ static void OptQuitXIns (const char* Opt attribute ((unused)),
 /* quit after MaxCycles cycles */
 {
     MaxCycles = strtoul(Arg, NULL, 0);
+    /* Guard against overflow. */
+    if (MaxCycles == ULONG_MAX && errno == ERANGE) {
+        Error("'-x parameter out of range. Max: %lu",ULONG_MAX);
+    }
 }
 
```

Comment:
mmmh, while this works - wouldnt it be better (as in simpler) to just make MaxCycles a uint64_t ? That'd also preemptively solve future overflow problems (we actually had a lot of these in VICE)

Reply:
or unsigned long long, which is guaranteed by C spec to be at least 64 bit, while having its own dedicated format string mnemonic "%llu" - using uint64_t requires macros like PRIu64 which requires hacks for MSVC.

Reply:
I feel this is not simply a review of the code linked above so I have transferred it to a discussion below.